src/Embedders/Word2Vec.php

```diff
@@ -0,0 +1,924 @@
+
+    /**
+     * Sigmoid activation function.
+     *
+     * @var \Rubix\ML\NeuralNet\ActivationFunctions\Sigmoid
+     */
+    protected $sigmoid;
+
+    /**
+     * @param string $layer
+     * @param int $window
+     * @param int $dimensions
+     * @param float $sampleRate
+     * @param float $alpha
+     * @param int $epochs
+     * @param int $minCount
+     * @throws \InvalidArgumentException
+     */
+    public function __construct(
+        string $layer = 'neg',
+        int $window = 2,
+        int $dimensions = 5,
+        float $sampleRate = 1e-3,
+        float $alpha = 0.01,
+        int $epochs = 10,
+        int $minCount = 2
+    ) {
+        if (!in_array($layer, ['neg', 'hs'])) {
+            throw new InvalidArgumentException('Layer must be neg or hs.');
+        }
+
+        if ($window > 5) {
+            throw new InvalidArgumentException("Window must be between 1 and 5, $window given.");
+        }
+
+        if ($dimensions < 5) {
+            throw new InvalidArgumentException("Dimensions must be greater than 4, $dimensions given.");
+        }
+
+        if ($sampleRate < 0.0) {
+            throw new InvalidArgumentException("Sample rate must be 0 or greater, $sampleRate given.");
+        }
+
+        if ($alpha < 0.0) {
+            throw new InvalidArgumentException("Alpha must be greater than 0, $alpha given.");
+        }
+
+        if ($epochs < 1) {
+            throw new InvalidArgumentException("Number of epochs must be greater than 0, $epochs given.");
+        }
+
+        if ($minCount < 1) {
+            throw new InvalidArgumentException("Minimum word count must be greater than 0, $minCount given.");
+        }
+
+        $this->layer = $layer;
+        $this->window = $window;
+        $this->dimensions = $dimensions;
+        $this->sampleRate = $sampleRate;
+        $this->alpha = $alpha;
+        $this->epochs = $epochs;
+        $this->minCount = $minCount;
+        $this->negLabels = Vector::quick([1, 0]);
+        $this->sigmoid = new Sigmoid();
+    }
+
+    /**
+     * Return the data types that this estimator is compatible with.
+     *
+     * @return \Rubix\ML\DataType[]
+     */
+    public function compatibility() : array
+    {
+        return [
+            DataType::categorical(),
+        ];
+    }
+
+    /**
+     * Return the settings of the hyper-parameters in an associative array.
+     *
+     * @return mixed[]
+     */
+    public function params() : array
+    {
+        return [
+            'layer' => $this->layer,
+            'window' => $this->window,
+            'dimensions' => $this->dimensions,
+            'sample_rate' => $this->sampleRate,
+            'alpha' => $this->alpha,
+            'epochs' => $this->epochs,
+            'min_count' => $this->minCount,
+        ];
+    }
+
+    /**
+     * Has the learner been trained?
+     *
+     * @return bool
+     */
+    public function fitted() : bool
+    {
+        return !empty($this->vectors);
```

Comment:
In the case that a user fits a dataset containing no text features (probably a mistake by the user but we have to still handle the situation) this would not be true. Instead, we could set initial state of the variable to null and check for a non-null value here instead.

Reply:
That's what it was originally set to but you requested that I change it :) 

Just to be clear, this will return false if the model has not been trained. So if they pass empty data, then it will return false.

Reply:
I added a 
`SamplesAreCompatibleWithTransformer::check($dataset, $this);` in fit method which will keep vectors null if they provided non text features.